src/Tribe/Migration/Queue.php

```diff
@@ -0,0 +1,287 @@
+	}
+
+	/**
+	 * Processes the next waiting batch of orders to migrate, if there are any.
+	 *
+	 * @since TBD
+	 *
+	 * @param null|int $batch_size Batch processing size override.
+	 */
+	public function process_queue( $batch_size = null ) {
+		if ( null === $batch_size ) {
+			$batch_size = $this->get_batch_size();
+		}
+
+		$current_offset = $this->get_current_offset();
+
+		if ( 'complete' === $current_offset ) {
+			$this->clear_scheduled_task();
+
+			return;
+		}
+
+		$this->batch_size = (int) $batch_size;
+
+		$processed = $this->process( $current_offset );
+
+		// if no items are processed or not processed clear the task.
+		if ( empty( $processed['processed'] ) && empty( $processed['not_processed'] ) ) {
+			tribe_update_option( $this->batch_offset_key, 'complete' );
+
+			$this->clear_scheduled_task();
+
+			return;
+		}
+
+		$this->update_offset( $processed['not_processed'] );
+	}
+
+	/**
+	 * Processes the next waiting batch of orders to migrate, if there are any.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $current_offset The current offset being used.
+	 *
+	 * @return array List of processed and not processed counts.
+	 */
+	public function process( $current_offset = 0 ) {
+		$counts = [
+			'processed'     => 0,
+			'not_processed' => 0,
+		];
+
+		$posts = $this->get_posts();
+
+		if ( empty( $posts ) ) {
+			return $counts;
+		}
+
+		foreach ( $posts as $post ) {
+			$processed = $this->process_post( $post );
+
+			if ( $processed ) {
+				$counts['processed'] ++;
+			} else {
+				$counts['not_processed'] ++;
+			}
+		}
+
+		return $counts;
+	}
+
+	/**
+	 * Process post in queue.
+	 *
+	 * @since TBD
+	 *
+	 * @param WP_Post|int $post The post object or post ID, depending on what $this->get_query() is set to return.
+	 *
+	 * @return bool Whether the post was processed.
+	 */
+	public function process_post( $post ) {
+		return false;
+	}
+
+	/**
+	 * Get posts for migrating.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $current_offset The current offset being used.
+	 *
+	 * @return int[]|WP_Post[] List of post IDs or post objects.
+	 */
+	public function get_posts( $current_offset ) {
+		return [];
+	}
+
+	/**
+	 * Get the Current offset number.
+	 *
+	 * @since TBD
+	 *
+	 * @return string|int Current offset number.
+	 */
+	public function get_current_offset() {
+		$current_offset = tribe_get_option( $this->batch_offset_key );
+
+		// Set up default current offset.
+		if ( false === $current_offset || '' === $current_offset ) {
+			$current_offset = 0;
+		}
+
+		return $current_offset;
+	}
+
+	/**
+	 * Update the Offset Number with the Current Batch.
+	 *
+	 * @since TBD
+	 *
+	 * @param int $not_processed the number of orders not processed.
+	 */
+	protected function update_offset( $not_processed ) {
+		$current_offset = $this->get_current_offset();
+
+		// Only set if numeric.
+		if ( ! is_numeric( $current_offset ) ) {
+			return;
+		}
+
+		$current_offset = (int) $current_offset;
+		$current_offset += $not_processed;
```

Comment:
:warning: Equals sign not aligned with surrounding assignments; expected 2 spaces but found 1 space

<sub>hash: 390e617f13cabadb30a3c7b4594f7428</sub>